code/game/objects/structures/platforms.dm

```diff
@@ -79,6 +93,49 @@
 			SPAN_WARNING("You finish dragging \the [user.pulling] onto \the [src]."))
 			user.pulling.forceMove(move_to_turf)
 
+/obj/structure/platform/attack_alien(mob/living/carbon/xenomorph/user)
+	if(stat & BROKEN)
+		to_chat(user, SPAN_WARNING("Its already destroyed!"))
+		return XENO_NO_DELAY_ACTION
+
+	if(stat & explo_proof)
+		to_chat(user, SPAN_WARNING("Its too strong for us!"))
+		return XENO_NO_DELAY_ACTION
+
+	if(user.action_busy)
+		return XENO_NO_DELAY_ACTION
+	if(user.a_intent != INTENT_DISARM)
+		user.set_interaction(src)
+		tgui_interact(user)
+		return XENO_ATTACK_ACTION
+	user.visible_message(SPAN_WARNING("[user] begins to lean against [src]."), \
+	SPAN_WARNING("You start to stomp and pressure [src]."), null, 5, CHAT_TYPE_XENO_COMBAT)
+	playsound(loc, creaking_sound, 30, 1)
+
+	var/shove_time
+	switch(user.mob_size)
+		if(MOB_SIZE_XENO_VERY_SMALL)
+			shove_time = 30 SECONDS
+		if(MOB_SIZE_XENO_SMALL)
+			shove_time = 9 SECONDS
+		if(MOB_SIZE_XENO)
+			shove_time = 7 SECONDS
+		if(MOB_SIZE_BIG)
+			shove_time = 4 SECONDS
+		if(MOB_SIZE_IMMOBILE)
+			shove_time = 2 SECONDS
+
+	xeno_attack_delay(user) //Adds delay here and returns nothing because otherwise it'd cause lag *after* finishing the shove.
+
+	if(!do_after(user, shove_time, INTERRUPT_ALL, BUSY_ICON_HOSTILE))
```

Comment:
Since this can be a long do_after, you need to check more often for whether an interrupt has occurred. 
```suggestion
	if(!do_after(user, shove_time, INTERRUPT_ALL, BUSY_ICON_HOSTILE, numticks = shove_time * 0.1))
```